src/Components/Panels/Toolbox/PassabilityControl.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { ChangeEvent } from 'react';
+
+import { IPassableObject } from '../../../Document/Attributes/IPassableObject';
+
+import styles from './PassabilityControls.module.scss';
+
+interface Props {
+  data: IPassableObject;
+  onChange: (changes: IPassableObject) => void;
+}
+
+const PassabilityControls = ({ data, onChange }: Props) => {
+  const handleDriveThroughOnChange = (_: ChangeEvent) => {
+    const drivethrough = !data.drivethrough;
+    onChange({
+      ...data,
+      drivethrough,
+    });
+  };
```

Comment:
```suggestion
  const handleDriveThroughOnChange = () => {
    onChange({
      ...data,
      drivethrough: !data.drivethrough,
    });
  };
```

If you're not going to use an argument, you can just omit it. Only use `_` if it's like a middle argument or it needs to be saved to a variable for some reason.

Also, no need for an intermediary variable. You only need an intermediary variable if it's doing some type of logic or is evaluating to a boolean value that isn't straightforward. e.g.

```js
const isShiftRequiredAndHeld = requirement.shift && event.shiftKey;
```

---

Same thing for shootthrough

Reply:
Inlining it like `drivethrough: !data.drivethrough` doesn't seem to work.